docs/reference/remap.cue

```diff
@@ -52,17 +51,589 @@ remap: {
+				.message = $name + " is a pretty great mascot" + " (though we're a bit biased)"
+				```
+				"""#
+			use: ["parameter", "return"]
+			examples: [
+				"\"I am a teapot\"",
+				#"""
+					"I am split
+					across multiple lines"
+					"""#,
+				#"This is not escaped, \"but this is\""#,
+			]
+		}
+		"timestamp": {
+			description: """
+				A string formatted as a timestamp. Timestamp specifiers can be either:
+
+				1. One of the built-in-formats listed in the [Timestamp Formats](#timestamp-formats)
+					table below, or
+				2. Any valid sequence of [time format specifiers](\(urls.chrono_time_formats)) from
+					Rust's `chrono` library.
+
+				### Timestamp Formats
+
+				The examples in this table are for 54 seconds after 2:37 am on December 1st, 2020 in
+				Pacific Standard Time.
+
+				Format | Description | Example
+				:------|:------------|:-------
+				`%F %T` | `YYYY-MM-DD HH:MM:SS` | `2020-12-01 02:37:54`
+				`%v %T` | `DD-Mmm-YYYY HH:MM:SS` | `01-Dec-2020 02:37:54`
+				`%FT%T` | [ISO 8601](\(urls.iso_8601))\\[RFC 3339](\(urls.rfc_3339)) format without time zone | `2020-12-01T02:37:54`
+				`%a, %d %b %Y %T` | [RFC 822](\(urls.rfc_822))/[2822](\(urls.rfc_2822)) without time zone | `Tue, 01 Dec 2020 02:37:54`
+				`%a %d %b %T %Y` | [`date`](\(urls.date)) command output without time zone | `Tue 01 Dec 02:37:54 2020`
+				`%a %b %e %T %Y` | [ctime](\(urls.ctime)) format | `Tue Dec  1 02:37:54 2020`
+				`%s` | [UNIX](\(urls.unix_timestamp)) timestamp | `1606790274`
+				`%FT%TZ` | [ISO 8601](\(urls.iso_8601))/[RFC 3339](\(urls.rfc_3339)) UTC | `2020-12-01T09:37:54Z`
+				`%+` | [ISO 8601](\(urls.iso_8601))/[RFC 3339](\(urls.rfc_3339)) UTC with time zone | `2020-12-01T02:37:54-07:00`
+				`%a %d %b %T %Z %Y` | [`date`](\(urls.date)) command output with time zone | `Tue 01 Dec 02:37:54 PST 2020`
+				`%a %d %b %T %z %Y`| [`date`](\(urls.date)) command output with numeric time zone | `Tue 01 Dec 02:37:54 -0700 2020`
+				`%a %d %b %T %#z %Y` | [`date`](\(urls.date)) command output with numeric time zone (minutes can be missing or present) | `Tue 01 Dec 02:37:54 -07 2020`
+				"""
+			use: ["parameter", "return"]
+			examples: [
+				"%a %d %b %T %Y",
+				"%FT%TZ",
+			]
+		}
+	}
+
+	// TRL syntax
+	#Operators: [_category=string]: [_op=string]: string
+
+	syntax: [RuleName=string]: {
+		#InNOut: {
+			in:  string
+			out: #RemapReturnTypes
+		}
+
+		name:        RuleName
+		href:        string // Ensures that we don't end up with clashing anchors
+		description: string
+		examples?: [string, ...string]
+		in_n_out?: [#InNOut, ...#InNOut]
+		operators?: #Operators
+		warnings?: [string, ...string]
+	}
+
+	syntax: {
+		"Paths": {
+			href: "paths"
+
+			description: """
+				In TRL, a dot (`.`) holds state across the script. At the beginning of the script,
+				it represents the object arriving into the transform; that object can be a log or a
+				metric. To give an example, imagine you're writing a TRL script to handle logs in
```

Comment:
Depending on where this documentation ends up, the object can be _anything_ the callee wants it to be, as long as they implement the relevant `Object` trait.

But specifically for Vector, yes, we have an `impl Object for LogEvent` and for `MetricEvent`, which allows TRL to mutate those two event types.